### PR TITLE
[Feature] Sync mmaction2-sdk(master) to dev1.x

### PR DESCRIPTION
**Pull request by irexyc**
## Motivation

sync https://github.com/open-mmlab/mmdeploy/pull/1183

### CODE REVIEW

File: demo/csrc/c/video_recognition.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+int main(int argc, char* argv[]) {
+  if (argc != 4) {
+    fprintf(stderr, "usage:\n  video_recognition device_name dump_model_directory video_path\n");
+    return 1;
+  }
+  auto device_name = argv[1];
+  auto model_path = argv[2];
+  auto video_path = argv[3];
+
+  int clip_len = 1;
```
Comment (grimoire) [inline demo/csrc/c/video_recognition.cpp:73]: Can we set these varibles by the input?
Or should we add some warning here?

**Review by grimoire — APPROVED** [review]
LGTM

**Review by lzhangzz — APPROVED** [review]

**Comment by RunningLeon** [comment]
@irexyc build_rknpu2_gcc failed

**Comment by lvhan028** [comment]
> @irexyc build_rknpu2_gcc failed

This is a known issue and will be fixed in #1343

**Pull request merged**